### ISSUE TITLE
Backup Camera Roll stuck at synchronisation

### ISSUE THREAD

**Issue by frederikb96** [scubajeff/lespas#107]
**Describe the bug**
After enabling camera roll backup in the settings, it starts to upload the pictures to the server and is able to create a respective folder on the server but gets stuck during upload and never finishes. No pictures are getting uploaded.

**To Reproduce**
- I freshly installed the App and connected to Nextcloud and used a new Test folder to contain the lespas folder
- I went into settings and enabled the camera roll backup slider and confirmed to upload the pictures (I tested it with different images)
- In synchronisation status,  it gets stuck at uploading the image, no process at all for one picture after one hour
- I checked  the nextcloud server site and the  DCIM folder gets created successfully but no images inside
- I can still upload images manually to new albums in the App and they all sync fine even though sync status is still stuck at the camera roll backup process
- The manual created albums also appear on the server site correctly

I also tried to reinstall the app again and use another clean lespas folder and used different images for upload or first having no images and enabling camera backup and then adding images afterwards but nothing helped and it always got stuck with the first upload for camera backup.

However,  everything else works perfectly and manually adding albums is all fine an  syncs instantly. 

**Expected behavior**
The camera backup runs successfully.

**Screenshots**
![Screenshot_20230308-113134_Les Pas.png](https://user-images.githubusercontent.com/83548283/223692661-34b0bfbc-bc81-4298-a30e-249f92875b05.png)

![Screenshot_20230308-112859_Les Pas.png](https://user-images.githubusercontent.com/83548283/223692667-a88a80c7-e3c2-4e8f-87db-73a9304753d9.png)

![Screenshot_20230308-112913_Les Pas.png](https://user-images.githubusercontent.com/83548283/223692662-55f0db84-20b7-4598-baf5-453941f4cae5.png)

![Screenshot_20230308-113211_Nextcloud.png](https://user-images.githubusercontent.com/83548283/223692664-c843bd5b-e60e-4f65-9396-f2eac4ef8161.png)

![Screenshot_20230308-113216_Nextcloud.png](https://user-images.githubusercontent.com/83548283/223692665-1852d975-909a-4da6-a4a4-7df88084dc5b.png)

![Screenshot_20230308-112808_Les Pas.png](https://user-images.githubusercontent.com/83548283/223692666-5e3ecab8-8f56-4e60-9153-cf108032a8b1.png)

![Screenshot_20230308-112938_Les Pas.png](https://user-images.githubusercontent.com/83548283/223692668-627e2009-c8f1-4018-8c72-cb9b55dc81a4.png)

**Smartphone (please complete the following information):**
- App version 2.7.3
 - Device: FP4
 - OS: /e/os 1.8
 - Version: Android 12
- Nextcloud AIO docker Setup version 25.0.4

**Additional context**
I can provide more information and logs if necessary :)

**Comment by frederikb96** [comment]
Also with single very small image no progress: 
![Screenshot_20230308-115844_Les Pas.png](https://user-images.githubusercontent.com/83548283/223695809-5756a0db-d422-4169-8d7a-6293dc65db4a.png)

![hu.jpg](https://user-images.githubusercontent.com/83548283/223695835-c56e8b4f-74b8-49f8-865d-0b58541f4698.jpg)

**Comment by scubajeff** [comment]
Frederik,

Thank you for reporting. If you can access your web server's access log, try grep 'LesPas' in it, find any line with response code 4xx and 5xx

**Comment by frederikb96** [comment]
Thank you very much for the fast response!

I have the Nextcloud AIO containers running and a caddy [reverse proxy setup as recommended for Nextcloud AIO](https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md).

In the caddy logs, I only found this line, even though there must have been many more requests failing, so not sure if this is a relevant one, at least it is about the upload to the DCIM folder, which never worked:
```
{"level":"error","ts":1678270934.170098,"logger":"http.log.error.log1","msg":"readfrom tcp 172.90.0.100:54644->172.90.0.1:11000: client disconnected","request":{"remote_ip":"172.90.0.1","remote_port":"58950","proto":"HTTP/2.0","method":"PUT","host":"cloud.bergrunde.net","uri":"/remote.php/dav/files/Freddy/Test/DCIM/fairphone_fp4/OpenCamera/Screenshot_20230308-103700_Lawnchair.png","headers":{"Authorization":[],"Content-Type":["image/png"],"Content-Length":["1449836"],"Accept-Encoding":["gzip"],"User-Agent":["LesPas_v2.7.3"]},"tls":{"resumed":false,"version":772,"cipher_suite":4865,"proto":"h2","server_name":"cloud.bergrunde.net"}},"duration":0.420701395,"status":502,"err_id":"hjahaz4p1","err_trace":"reverseproxy.statusError (reverseproxy.go:1281)"}
```

I also tested the upload function with an entirely different android smartphone running Android 8, and same problem here with my Nextcloud server.

**Comment by frederikb96** [comment]
Hm, now the app got stuck in this sync local changes mode, even though camera roll is disabled. Force stop and clearing cache did not help.

**Comment by scubajeff** [comment]
http response code 502 means Back Gateway, you should somehow check your web server's configuration. Can I access your server remotely? It's more easier to debug in this way.

**Comment by frederikb96** [comment]
Sorry, I guess I do not feel comfortable giving someone access to my server, so no problem if it is too tiresome for you to debug this issue, but thank you anyway :)

However if you feel like debugging, I have just analyzed the "real" caddy log (previously it was via `docker log`) and now I can see all the requests and errors that have occurred regarding lespas. I did a new fresh installation of lespas and removed every trace of lespas from my server and started the logging with the new first launch of the app on my phone. I mailed you the complete web server log containing only all entries related to lespas from this session.

I got the logs via `tail -f cloud.log | grep -ie "lespas" | jq > lespas.log`.

I tested the following:
- First, in the app, I uploaded a picture manually to a new album "test" and this worked
- Then, I took a photo with my camera and enabled in the settings the camera upload function and started a sync - gets stuck
- I went back to main view and uploaded another image to the "test" album, worked again
- I disabled the backup function again

Here are some errors, which might be interesting with status 404 (not found) and 409 (conflict)
<details><summary>Click to expand</summary><p>

```
{
  "level": "error",
  "ts": 1678290301.119089,
  "logger": "http.log.access.log1",
  "msg": "handled request",
  "request": {
    "remote_ip": "172.90.0.1",
    "remote_port": "50830",
    "proto": "HTTP/2.0",
    "method": "PROPFIND",
    "host": "cloud.bergrunde.net",
    "uri": "/remote.php/dav/files/Freddy/Test/DCIM/fairphone_fp4",
    "headers": {
      "Cache-Control": [
        "no-cache"
      ],
      "Depth": [
        "0"
      ],
      "Authorization": [],
      "Accept-Encoding": [
        "gzip"
      ],
      "User-Agent": [
        "LesPas_v2.7.3"
      ]
    },
    "tls": {
      "resumed": false,
      "version": 772,
      "cipher_suite": 4865,
      "proto": "h2",
      "server_name": "cloud.bergrunde.net"
    }
  },
  "user_id": "",
  "duration": 0.126614073,
  "size": 252,
  "status": 404,
  "resp_headers": {
    "Vary": [
      "Brief,Prefer"
    ],
    "X-Xss-Protection": [
      "1; mode=block"
    ],
    "Cache-Control": [
      "no-store, no-cache, must-revalidate"
    ],
    "X-Powered-By": [
      "PHP/8.0.28"
    ],
    "X-Content-Type-Options": [
      "nosniff"
    ],
    "Alt-Svc": [
      "h3=\":443\"; ma=2592000"
    ],
    "Pragma": [
      "no-cache"
    ],
    "Set-Cookie": [],
    "Strict-Transport-Security": [
      "max-age=31536000;",
      "max-age=31536000;"
    ],
    "Content-Security-Policy": [
      "default-src 'none';"
    ],
    "Expires": [
      "Thu, 19 Nov 1981 08:52:00 GMT"
    ],
    "X-Robots-Tag": [
      "none"
    ],
    "X-Frame-Options": [
      "SAMEORIGIN"
    ],
    "Referrer-Policy": [
      "no-referrer"
    ],
    "Server": [
      "Caddy",
      "Caddy",
      "Apache/2.4.55 (Unix)"
    ],
    "Date": [
      "Wed, 08 Mar 2023 15:45:00 GMT"
    ],
    "Dav": [
      "1, 3, extended-mkcol, access-control, calendarserver-principal-property-search, nextcloud-checksum-update, nc-calendar-search, nc-enable-birthday-calendar"
    ],
    "Content-Type": [
      "application/xml; charset=utf-8"
    ],
    "X-Permitted-Cross-Domain-Policies": [
      "none"
    ]
  }
}
{
  "level": "error",
  "ts": 1678290301.7107995,
  "logger": "http.log.access.log1",
  "msg": "handled request",
  "request": {
    "remote_ip": "172.90.0.1",
    "remote_port": "50830",
    "proto": "HTTP/2.0",
    "method": "PUT",
    "host": "cloud.bergrunde.net",
    "uri": "/remote.php/dav/files/Freddy/Test/DCIM/fairphone_fp4/OpenCamera/IMG_20230308_164241.jpg",
    "headers": {
      "Authorization": [],
      "Content-Type": [
        "image/jpeg"
      ],
      "Content-Length": [
        "581825"
      ],
      "Accept-Encoding": [
        "gzip"
      ],
      "User-Agent": [
        "LesPas_v2.7.3"
      ]
    },
    "tls": {
      "resumed": false,
      "version": 772,
      "cipher_suite": 4865,
      "proto": "h2",
      "server_name": "cloud.bergrunde.net"
    }
  },
  "user_id": "",
  "duration": 0.317427158,
  "size": 181,
  "status": 409,
  "resp_headers": {
    "Server": [
      "Caddy",
      "Caddy",
      "Apache/2.4.55 (Unix)"
    ],
    "X-Xss-Protection": [
      "1; mode=block"
    ],
    "Referrer-Policy": [
      "no-referrer"
    ],
    "X-Frame-Options": [
      "SAMEORIGIN"
    ],
    "Strict-Transport-Security": [
      "max-age=31536000;",
      "max-age=31536000;"
    ],
    "Pragma": [
      "no-cache"
    ],
    "Content-Type": [
      "application/xml; charset=utf-8"
    ],
    "X-Robots-Tag": [
      "none"
    ],
    "X-Powered-By": [
      "PHP/8.0.28"
    ],
    "X-Content-Type-Options": [
      "nosniff"
    ],
    "Cache-Control": [
      "no-store, no-cache, must-revalidate"
    ],
    "Expires": [
      "Thu, 19 Nov 1981 08:52:00 GMT"
    ],
    "Alt-Svc": [
      "h3=\":443\"; ma=2592000"
    ],
    "Set-Cookie": [],
    "Date": [
      "Wed, 08 Mar 2023 15:45:01 GMT"
    ],
    "X-Permitted-Cross-Domain-Policies": [
      "none"
    ],
    "Content-Security-Policy": [
      "default-src 'none';"
    ]
  }
}
```

</p></details>

And I am wondering, why lespas is trying to access this folder, which is not in my lespas folder and failing to access a .json file here which is not existent?
<details>
<summary>Click to expand</summary>

```
{
  "level": "error",
  "ts": 1678289954.6095674,
  "logger": "http.log.access.log1",
  "msg": "handled request",
  "request": {
    "remote_ip": "172.90.0.1",
    "remote_port": "32928",
    "proto": "HTTP/2.0",
    "method": "GET",
    "host": "cloud.bergrunde.net",
    "uri": "/remote.php/dav/files/Freddy/Pictures/00_Handy/2023/2023_00_Freunde_N/791225.json",
    "headers": {
      "Cache-Control": [
        "no-cache"
      ],
      "Authorization": [],
      "Accept-Encoding": [
        "gzip"
      ],
      "User-Agent": [
        "LesPas_v2.7.3"
      ]
    },
    "tls": {
      "resumed": false,
      "version": 772,
      "cipher_suite": 4865,
      "proto": "h2",
      "server_name": "cloud.bergrunde.net"
    }
  },
  "user_id": "",
  "duration": 0.146007781,
  "size": 281,
  "status": 404,
  "resp_headers": {
    "Expires": [
      "Thu, 19 Nov 1981 08:52:00 GMT"
    ],
    "X-Xss-Protection": [
      "1; mode=block"
    ],
    "X-Content-Type-Options": [
      "nosniff"
    ],
    "Referrer-Policy": [
      "no-referrer"
    ],
    "Date": [
      "Wed, 08 Mar 2023 15:39:14 GMT"
    ],
    "X-Powered-By": [
      "PHP/8.0.28"
    ],
    "X-Frame-Options": [
      "SAMEORIGIN"
    ],
    "Content-Type": [
      "application/xml; charset=utf-8"
    ],
    "Alt-Svc": [
      "h3=\":443\"; ma=2592000"
    ],
    "Set-Cookie": [],
    "Cache-Control": [
      "no-store, no-cache, must-revalidate"
    ],
    "Content-Security-Policy": [
      "default-src 'none';"
    ],
    "Strict-Transport-Security": [
      "max-age=31536000;",
      "max-age=31536000;"
    ],
    "X-Permitted-Cross-Domain-Policies": [
      "none"
    ],
    "X-Robots-Tag": [
      "none"
    ],
    "Pragma": [
      "no-cache"
    ],
    "Server": [
      "Caddy",
      "Caddy",
      "Apache/2.4.55 (Unix)"
    ]
  }
}
```

</details>

As recommended by Nextcloud, I use the following reverse proxy setup with caddy:
172.90.0.1 is my docker network gateway, and port 11000 is the one from the automatically created Nextcloud AIO Apache container.
```
cloud.bergrunde.net {
	tls {$EMAIL}
	
	header Strict-Transport-Security max-age=31536000;
	
	reverse_proxy 172.90.0.1:11000
	
	log {
		output file /var/log/caddy/cloud.log {
		}
	}
	
}
```

**Comment by scubajeff** [comment]
With that much of detail you provided, a direct access won't be necessary.

I read the log you sent me. Found interesting difference how Caddy response to webdav PUT calls as to the other web servers, When Les Pas uploading a new file `IMG_20230308_164241.jpg` to a not existing folder `.../Test/DCIM/fairphone_fp4/`**`OpenCamera`**`/`, instead of a response code 404 like Apache does, it spills 409. That's why the camera roll backup process stuck.

Debug build with the fix is attached [here](https://github.com/scubajeff/lespas/files/10926687/LesPas-v2.7.5-debug-Caddy-409.zip). Could you please give it a try?

As for why Les Pas was peeking outside of `lespas/` folder, it's because those are folders you've shared with others or shares you received from others, Les Pas was figuring out whether the share was published by itself or not by looking for a json file in it.

**Comment by frederikb96** [comment]
You are awesome, it is fixed with this new version! I did the same procedure as last time and, just for completeness, mailed you the logs again.
And thanks for your explanation with the "peeking", that makes sense now :)

Btw: Does this mean, that if I share a folder via Les Pas and on the receipts site, I move it to a different location on the server side, Les Pas will still find it and progress it normally. That is actually quite nice, such that one is not obligated to keep all the pictures within the shared_with_me folder :)

Btw2: It looks like the F-Droid version is behind the play store version. Do you know the reason? Probably due to F-Droid checking the new releases, right?

I will close the issue than, if you need more logs for debugging, let me know. Thanks a lot!

**Comment by frederikb96** [comment]
Looks lile #106 is related and might also be fixed with this.

**Comment by scubajeff** [comment]
No, the folder should stay in `share_with_me`, otherwise, things will break.

F-Droid pulls repo and builds on their own machine to make sure there's no modification to the source. That's why it's always several days behind.

You were using r2.7.3. r2.7.4 fixed #106, the latest is actually r.2.7.5 and fix to this issue will be included in upcoming r2.7.6
